### DISCUSSION TITLE
You must have a "project.xml" file or specify another valid project file when using the 'test' command

### DISCUSSION THREAD

**Discussion by ethancolon** [KadeArchive/Kade-Engine#1081] [Q&A]
i have is bug when i try build a game or do modchart i get is
![save me](https://user-images.githubusercontent.com/67854275/123711775-05655b80-d83f-11eb-956c-0d69cebbc30e.PNG)

**Comment by ACardboardBox1** [comment]
click on the folders that appear until you see project.xml. copy the directory with project.xml, cd to it, and then lime build windows (or lime test windows)

**Comment by ljharnish** [comment]
You're in the assets folder, you need to be in the second Kade-Engine-1.4.2 folder.

**Reply by v4lcyxn** [reply]
Can you help me please?

**Reply by LaughingMoon8** [reply]
There is no project.xml in the bin is the problem.

**Reply by HaxLua** [reply]
just do cd ..

**Reply by OptiC992** [reply]
exit out of the kade engine folder, re enter it and when you enter it press kade engine-1.4.2 then go to power shell

**Comment by RealDealJinx** [comment]
What i do is i go where you see like source,assests,art,etc i click refresh and it works

**Comment by raf13lol** [comment]
do cd C:\Users\Allstar Auto\Downloads\Kade-Engine-1.4.2\Kade-Engine-1.4.2

then lime test windows

**Comment by TartIsAnEpicChad** [comment]
i am having the problem too, but i dont get what "copy the directory with project.xml" means.
could you write down what i have to put?

**Reply by ghost** [reply]
cd C: \Users\AllStar Auto\Desktop\Kade-Engine-1.4. 2\Kade-Engine-1.4.2\

**Reply by vaasans** [reply]
NOT WORKING

**Reply by OptiC992** [reply]
exit out of the kade engine folder, re enter it and when you enter it press kade engine-1.4.2 then go to power shell

**Comment by NK-523** [comment]
Click off the assets folder, or go back one folder from the directory and click your kade engine folder, the run the command.

**Reply by LaughingMoon8** [reply]
its shows the statePointer error

**Reply by NK-523** [reply]
run these commands in command prompt 
`haxelib remove linc_luajit`
`haxelib remove hxvm-luajit`
`haxelib git linc_luajit https://github.com/nebulazorua/linc_luajit.git `
`haxelib git hxvm luajit https://github.com/nebulazorua/hxvm-luajit`

**Reply by EyeQueu3** [reply]
haxelib git hxvm luajit https://github.com/nebulazorua/hxvm-luajit isn't working, it says  "Error: Could not clone Git repository".

**Reply by Reycko** [reply]
try `haxelib git hxvm-luajit https://github.com/nebulazorua/hxvm-luajit` instead

**Comment by Jotaro-Gaming** [comment]
> i have is bug when i try build a game or do modchart i get is
> ![save me](https://user-images.githubusercontent.com/67854275/123711775-05655b80-d83f-11eb-956c-0d69cebbc30e.PNG)

sussy baka ur in the assets folder just put it in the source code

**Reply by OptiC992** [reply]
exit out of the kade engine folder, re enter it and when you enter it press kade engine-1.4.2 then go to power shell

**Comment by Primezen** [comment]
![image](https://user-images.githubusercontent.com/92563036/137593659-b4af5143-69c3-4d63-8cc9-d940d7319fb6.png)

i have the project file but like i still have the same problem idk why

**Reply by Primezen** [reply]
![image](https://user-images.githubusercontent.com/92563036/137593703-64afce8a-5e83-43d2-8adf-464aef013a08.png)

the command btw

**Reply by The1Unkn0wn** [reply]
bruh same issue help me

**Reply by OptiC992** [reply]
exit out of the kade engine folder, re enter it and when you enter it press kade engine-1.4.2 then go to power shell

**Comment by Jotaro-Gaming** [comment]
make it show the ends (like project.xml

**Reply by Primezen** [reply]
still doesn't work somehow

**Reply by Jotaro-Gaming** [reply]
> still doesn't work somehow

wait did you install Visual community 2019??

**Reply by ZakwanArief** [reply]
Yea im currently using Visual Community 2019
and this is an error I got
![image](https://user-images.githubusercontent.com/93035165/138552194-2217872f-c9be-42b2-9239-26c59fc096b1.png)

**Comment by The1Unkn0wn** [comment]
> i have is bug when i try build a game or do modchart i get is ![save me](https://user-images.githubusercontent.com/67854275/123711775-05655b80-d83f-11eb-956c-0d69cebbc30e.PNG)

I had the same issue, but I used the folder after clicking in once, and using the folder I have in my downloads. If it matters, I use Windows PowerShell, as I couldn't figure out how to type the code in Visual Studio.

**Reply by raf13lol** [reply]
1st. use vsc not vs
2nd. do 
```powershell
 cd..
```
then try again

**Reply by The1Unkn0wn** [reply]
nvm i found the project file XD I just skimmed by it, if it still has issues i'll try yours

**Reply by The1Unkn0wn** [reply]
excuse me?
![image](https://user-images.githubusercontent.com/93896534/143720621-d57c627e-2338-488d-8f01-78a93198f37a.png)

**Reply by The1Unkn0wn** [reply]
Well, I couldn't figure this out, and the instructions in the files and on here, this is my first time really coding anything, I'm honestly so confused, why can't I figure out how to get a "bin" or something. :(

**Reply by raf13lol** [reply]
ok do these commands
```cmd
cd "C:\Users\djb05\Downloads\Melon Man Mod\Kade-Engine-1.8\"
lime test windows
```

**Reply by The1Unkn0wn** [reply]
OK I'll try it

**Reply by The1Unkn0wn** [reply]
I think that fixed it I just gotta remove the weeks that are not being used

**Reply by The1Unkn0wn** [reply]
I think I just put in more work for myself by removing all weeks other than tutorial, Week 1, and Week 2 :/

**Reply by The1Unkn0wn** [reply]
> ok do these commands
> 
> ```batchfile
> cd "C:\Users\djb05\Downloads\Melon Man Mod\Kade-Engine-1.8\"
> lime test windows
> ```

I did that, I had to fix something, but then...
![image](https://user-images.githubusercontent.com/93896534/143779869-adb4d97e-d7f1-43e3-b650-401f6941d10b.png)

**Reply by raf13lol** [reply]
https://github.com/KadeDev/Kade-Engine/blob/stable/docs/troubleshooting.md#error-source-path-chaxetoolkithaxelibextension-webmgitndllwindows64extension-webmndll-does-not-exist

**Reply by The1Unkn0wn** [reply]
same issue
![image](https://user-images.githubusercontent.com/93896534/143791118-b8bebd63-53b8-48db-a9d4-81ef484b2dfb.png)

**Reply by OptiC992** [reply]
you have to download the libraries

**Reply by OptiC992** [reply]
look

**Reply by OptiC992** [reply]
![Capture](https://user-images.githubusercontent.com/95190039/143794048-673ced67-06b8-4847-b1dd-b266b2304885.PNG)

**Reply by The1Unkn0wn** [reply]
> https://github.com/KadeDev/Kade-Engine/blob/stable/docs/troubleshooting.md#error-source-path-chaxetoolkithaxelibextension-webmgitndllwindows64extension-webmndll-does-not-exist

That let me test it, but how do i turn it into a .exe file?

**Reply by The1Unkn0wn** [reply]
> https://github.com/KadeDev/Kade-Engine/blob/stable/docs/troubleshooting.md#error-source-path-chaxetoolkithaxelibextension-webmgitndllwindows64extension-webmndll-does-not-exist

Also, when I tested it, the volume was busted so that the sound for changing it always happens, I'm not sure how or why

**Comment by NonCanonBF** [comment]
>

**Reply by NonCanonBF** [reply]
To make a directory, the easiest way is to open the file and go to windows PowerShell from the directory you want that reaches the project.xml file. To open windows PowerShell from the file explorer, click the file tab in the file explorer and click "windows PowerShell".
